lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -118,6 +118,25 @@ sub _search_perl_modules {
     return \@results;
 }
 
+sub _search_job_modules {
+    my ($self, $keywords, $limit) = @_;
+
+    my @results;
+    my $last_job    = -1;
+    my $like        = {like => "%${keywords}%"};
+    my $job_modules = $self->schema->resultset('JobModules')->search({-or => {script => $like}})->slice(0, $limit);
```

Comment:
You likely want to join with the jobs table right away to sort these by the most recent job. On the production instances this query alone would likely already fill up the general `$limit`. Maybe it would make sense to divide the `$limit` so you get a little bit of each search category.

Reply:
> From Rocket.Chat
> Maybe it is easier to query the jobs and join them by the job modules. sub complex_query already does that filtering for certain module results. Your code would be similar but you would filter for the module name (instead of the result).
Looking into that.

I would refrain from extending the concept of the limit here. As per @Amrysliu 's point I changed it to only consider the `name`.